naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionField.scala

```diff
@@ -48,8 +50,13 @@ object NaptimeUnionField {
         case definitions: java.util.Map[String @unchecked, String @unchecked] => definitions.asScala
       }.getOrElse(Map[String, String]())
 
-      val unionMemberKey = typedDefinitions
-        .getOrElse(subType.getUnionMemberKey, subType.getUnionMemberKey)
+      val unionMemberKey = subType match {
+        case sType if typedDefinitions.contains(sType.getUnionMemberKey) =>
```

Comment:
we actually don't need to assign `subType` back to anything, since we're not changing its type or anything. so we can actually do:
```scala
case _ if typedDefinitions.contains(subType.getUnionMemberKey) =>
  typedDefinitions(subType.getUnionMemberKey
```